src/solver/modelConverter/convertorVisitor.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include <iostream>
+
+#include <antares/solver/modelConverter/convertorVisitor.h>
+
+#include "ExprLexer.h"
+#include "ExprParser.h"
+#include "antlr4-runtime.h"
+
+namespace Antares::Solver::ModelConverter
+{
+
+using namespace Antares::Solver::Nodes;
+
+static Node* toNodePtr(const std::any& a)
+{
+    return std::any_cast<Node*>(a);
+}
+
+Node* convertExpressionToNode(const std::string& exprStr,
+                              Antares::Solver::Registry<Node>& registry,
+                              const ModelParser::Model& model)
+{
+    if (exprStr.empty())
+    {
+        return nullptr;
+    }
+
+    antlr4::ANTLRInputStream input(exprStr);
+    ExprLexer lexer(&input);
+    antlr4::CommonTokenStream tokens(&lexer);
+    ExprParser parser(&tokens);
+
+    ExprParser::ExprContext* tree = parser.expr();
+
+    ConvertorVisitor visitor(registry, model);
+    return toNodePtr(visitor.visit(tree));
+}
+
+ConvertorVisitor::ConvertorVisitor(Antares::Solver::Registry<Node>& registry,
+                                   const ModelParser::Model& model):
+    registry_(registry),
+    model_(model)
+{
+}
+
+std::any ConvertorVisitor::visit(antlr4::tree::ParseTree* tree)
+{
+    return tree->accept(this);
+}
+
+std::any ConvertorVisitor::visitIdentifier(ExprParser::IdentifierContext* context)
+{
+    bool is_parameter = false;
+    for (const auto& param: model_.parameters)
+    {
+        if (param.id == context->getText())
+        {
+            is_parameter = true;
+            break;
+        }
+    }
+
+    return (is_parameter) ? static_cast<Node*>(registry_.create<ParameterNode>(context->getText()))
+                          : static_cast<Node*>(registry_.create<VariableNode>(context->getText()));
```

Comment:
in case it is not a parameter, I think it would be better to check that it is a variable indeed. else, throw an exception. this will make the model parsing throw at parse time instead of at runtime